community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ProcedureHolder.java

```diff
@@ -36,11 +38,12 @@
 class ProcedureHolder<T>
 {
     private final Map<QualifiedName,Integer> nameToId = new HashMap<>();
+    private final Map<QualifiedName,Integer> caseInsensitveName2Id = new HashMap<>();
```

Comment:
It feels like this should be able to be done without the extra map, but we can leave that as a future improvement